cmd/kubelet/kubelet.go

```diff
@@ -93,9 +93,10 @@ func main() {
 
 	// use kubeletServer to construct the default KubeletDeps
 	kubeletDeps, err := app.UnsecuredDependencies(kubeletServer)
-
-	// add the kubelet config controller to kubeletDeps
-	kubeletDeps.KubeletConfigController = kubeletConfigController
+	if err == nil {
```

Comment:
Good catch @yiqinguo.
I agree with @yujuhong, we need to actually handle the error by killing the Kubelet.